apps/crn-server/test/integration/reminders.integration-test.ts

```diff
@@ -201,38 +224,42 @@ describe('Reminders', () => {
         researchOutputInput3,
       );
 
-      const reminders = await reminderDataProvider.fetch(fetchRemindersOptions);
-      const reminderIds = reminders.items.map(
-        (item) =>
-          (item as ResearchOutputPublishedReminder).data.researchOutputId,
-      );
+      await retryable(async () => {
+        const reminders = await reminderDataProvider.fetch(
+          fetchRemindersOptions,
+        );
+        const reminderIds = reminders.items.map(
+          (item) =>
+            (item as ResearchOutputPublishedReminder).data.researchOutputId,
+        );
 
-      expect(reminderIds).toEqual([
-        researchOutputId2,
-        researchOutputId3,
-        researchOutputId1,
-      ]);
+        expect(reminderIds).toEqual([
+          researchOutputId2,
+          researchOutputId3,
+          researchOutputId1,
+        ]);
+      });
     });
   });
 
   describe('Event Happening Today Reminder', () => {
     let userId: string;
+    let teamId: string;
     let calendarId: string;
-    let eventIdsForDeletion: string[] = [];
     let fetchRemindersOptions: FetchRemindersOptions;
 
-    beforeAll(async () => {
+    beforeEach(async () => {
       jest.useFakeTimers();
 
       const teamCreateDataObject = getTeamCreateDataObject();
       teamCreateDataObject.applicationNumber = chance.name();
-      const teamId = await teamDataProvider.create(teamCreateDataObject);
+      teamId = await teamDataProvider.create(teamCreateDataObject);
 
       const userCreateDataObject = getUserInput(teamId);
       userId = await userDataProvider.create(userCreateDataObject);
     });
 
-    afterAll(() => {
+    afterEach(async () => {
       jest.useRealTimers();
```

Comment:
yeah i would probably extract the timers into a beforeAll and beforeEach

Also by running the setup each time for every test we are slowing it down quite significantly (I think you have already extended jest's timeout here to something really high). As far as I remember the team and user fixtures were reused between the tests - maybe we should consider tearing down just the data we know we need to purge after every tests?

Although I admit this would be dangerous, as we could accidentally run into the same problems we had before, plus there are some user and team fixtures down below that need to be torn down just as well, so yeah it's really just a hint in case it's got too slow that maybe we want to be more selective about which data we want to keep between the tests

Reply:
I bumped the timer from 30s to 2 minutes. The actual test suite takes a little over 60 seconds to run now, but was occasionally timing out before when set at 30 seconds, so would need increasing anyway. 2 minutes seemed like a safe margin but without being excessively slow.

I agree it is sub-optimal, but given it's running in parallel with the jest unit test runners - all of which are currently taking a good time longer than this task still - and given the need for manual teardown otherwise, then I'd suggest leaving it as it is.

If you feel strongly about it I'm happy to revisit.

Reply:
I will pull out the timers though.

Reply:
Updated with the timers moved into `beforeAll`

I've run a few test runs locally and the typical execution time is 40-45 seconds, but the timeout should be a good deal longer to handle the possibility that tests may need to retry, hence two minutes.